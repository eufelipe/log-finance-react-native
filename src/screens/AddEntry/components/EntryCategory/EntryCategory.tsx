import React from 'react';

import {useNavigation} from '@react-navigation/native';
import {useTranslation} from 'react-i18next';
import {CategoriesNavigationProp} from 'routes/StacksRoute';

import {CategoryIcon} from 'components';
import Row from '../Row';

import {Label, Title, Touchable} from './styles';

const EntryCategory = (): JSX.Element => {
  const navigation = useNavigation<CategoriesNavigationProp>();

  const {t} = useTranslation('add');

  const handleCategories = () => {
    navigation.navigate('CategoriesStack');
  };

  return (
    <Row>
      <Label>{t('category')}</Label>
      <Touchable onPress={handleCategories}>
        <CategoryIcon contained />
        <Title>{t('category-default')}</Title>
      </Touchable>
    </Row>
  );
};

export default EntryCategory;
