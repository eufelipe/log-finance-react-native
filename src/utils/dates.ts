import {
  format as formatPattern,
  isValid,
  parseISO as parseISOFNS,
} from 'date-fns';

import locale from 'date-fns/locale/pt-BR';

export const DATE_FORMAT_ISO = 'yyyy-MM-dd';
export const DATE_FORMAT_BR = 'dd/MM/yyyy';
export const DATE_FORMAT_FRIENDLY = " dd/MM 'às' HH:mm";
export const DATE_FORMAT_FRIENDLY_TIME = "'às' HH:mm";

export function parseStringToDate(value: string): Date {
  const parsedDate = parseISOFNS(value);
  if (!isValid(parsedDate)) {
    throw Error('canot parse date');
  }
  return parsedDate;
}
export function parseDateToString({
  value,
  format = DATE_FORMAT_BR,
}: {
  value: Date;
  format?: string;
}): string {
  return formatPattern(value, format, {locale});
}

export const getDate = (date: Date): string => {
  return formatPattern(date, DATE_FORMAT_ISO, {locale});
};

export const getDateToday = (): string => {
  const today = new Date();
  return formatPattern(today, DATE_FORMAT_ISO, {locale});
};

export const getDateAndHourFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY, {locale});
};

export const getTimeFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY_TIME, {locale});
};
