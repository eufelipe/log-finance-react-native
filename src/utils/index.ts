export {default as number} from './number';
export {default as strings} from './strings';
