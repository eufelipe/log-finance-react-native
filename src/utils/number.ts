export const sanitizeToFloat = (value: string): number => {
  value = value.replace(/[^0-9.]+/, '');
  if (value.includes('.') && value.includes(',')) {
    value = value.replace(/\./g, '').replace(',', '.');
  } else if (value.includes(',')) {
    value = value.replace(',', '.');
  }
  return parseFloat(value) || 0;
};
