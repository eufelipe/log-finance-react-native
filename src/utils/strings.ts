export const sanitizeString = (value: string): string => {
  return value
    .toLowerCase()
    .replace(/á/g, 'a')
    .replace(/é/g, 'e')
    .replace(/í/g, 'i')
    .replace(/ó/g, 'o')
    .replace(/ú/g, 'u');
};
