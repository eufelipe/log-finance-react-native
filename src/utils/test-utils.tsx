/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render, RenderAPI, RenderOptions} from '@testing-library/react-native';

import {theme} from 'styles/theme';
import {ThemeProvider} from 'styled-components/native';

import {NavigationContainer} from '@react-navigation/native';

interface Props {
  children: JSX.Element;
}

const AllTheProviders = ({children}: Props) => {
  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer>{children}</NavigationContainer>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions,
): RenderAPI => render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
