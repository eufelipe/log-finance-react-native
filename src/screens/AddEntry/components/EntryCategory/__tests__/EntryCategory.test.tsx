/* eslint-disable @typescript-eslint/unbound-method */
/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render} from 'utils/test-utils';
import EntryCategory from '../EntryCategory';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({t: (key: any) => key}),
}));

describe('EntryCategory', () => {
  it('Test match snapshot EntryCategory', () => {
    // given
    const props = {};

    // when
    const {toJSON} = render(<EntryCategory {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
