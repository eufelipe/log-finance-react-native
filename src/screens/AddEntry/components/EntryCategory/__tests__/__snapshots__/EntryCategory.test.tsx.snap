// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryCategory Test match snapshot EntryCategory 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "#E1E3E510",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "marginBottom": 0,
        "marginLeft": 42.857142857142854,
        "marginRight": 42.857142857142854,
        "marginTop": 21.428571428571427,
        "paddingBottom": 42.857142857142854,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff50",
            "fontFamily": "Poppins-Regular",
            "fontSize": 28,
            "fontWeight": "400",
          },
        ]
      }
    >
      category
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 5,
          "opacity": 1,
        }
      }
    >
      <View
        color="#6F7D9B"
        large={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#6F7D9B",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 24,
              },
              Array [
                Object {},
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î¢–
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Poppins-Regular",
              "fontSize": 28,
              "fontWeight": "700",
              "paddingLeft": 21.428571428571427,
            },
          ]
        }
      >
        category-default
      </Text>
    </View>
  </View>
</View>
`;
